planner/core/expression_rewriter.go

```diff
@@ -1631,35 +1631,46 @@ func (er *expressionRewriter) patternLikeToExpression(v *ast.PatternLikeExpr) {
 	var function expression.Expression
 	fieldType := &types.FieldType{}
 	isPatternExactMatch := false
+	escape := "\\"
+	if escapeExpression, ok := er.ctxStack[l-1].(*expression.Constant); ok {
+		str, isNull, err := escapeExpression.EvalString(nil, chunk.Row{})
+		if err != nil {
+			er.err = err
+			return
+		}
+		if !isNull {
+			escape = str
+		}
+	}
 	// Treat predicate 'like' the same way as predicate '=' when it is an exact match and new collation is not enabled.
-	if patExpression, ok := er.ctxStack[l-1].(*expression.Constant); ok && !collate.NewCollationEnabled() {
+	if patExpression, ok := er.ctxStack[l-2].(*expression.Constant); ok && !collate.NewCollationEnabled() {
 		patString, isNull, err := patExpression.EvalString(nil, chunk.Row{})
 		if err != nil {
 			er.err = err
 			return
 		}
 		if !isNull {
-			patValue, patTypes := stringutil.CompilePattern(patString, v.Escape)
-			if stringutil.IsExactMatch(patTypes) && er.ctxStack[l-2].GetType().EvalType() == types.ETString {
+			patValue, patTypes := stringutil.CompilePattern(patString, escape[0])
+			if stringutil.IsExactMatch(patTypes) && er.ctxStack[l-l].GetType().EvalType() == types.ETString {
 				op := ast.EQ
 				if v.Not {
 					op = ast.NE
 				}
 				types.DefaultTypeForValue(string(patValue), fieldType, char, col)
-				function, er.err = er.constructBinaryOpFunction(er.ctxStack[l-2],
+				function, er.err = er.constructBinaryOpFunction(er.ctxStack[l-3],
 					&expression.Constant{Value: types.NewStringDatum(string(patValue)), RetType: fieldType},
 					op)
 				isPatternExactMatch = true
 			}
 		}
 	}
 	if !isPatternExactMatch {
```

Comment:
we need also to consider the parameter maker.

Reply:
I made fixed to avoid parameter marker in rewrite to "=“ logic, please check to see if its right, THKS